src/transformers/modeling_utils.py

```diff
@@ -1442,6 +1449,9 @@ def post_init(self):
         """
         self.init_weights()
         self._backward_compatibility_gradient_checkpointing()
+        # If current model is a base model, attach `base_model_tp_plan` from config
+        if self.base_model is self:
```

Comment:
```suggestion
        if self._tp_plan is None:
```

Reply:
feels simpler

Reply:
Yeah, but `self._tp_plan` should be None only for the base model no? 
We could maybe add something like `if not self.base_model is self and self._tp_plan is None and self.supports_tp_plan` raise an error in the futur, to force people to add the TP plan.

Reply:
> but self._tp_plan should be None only for the base model no?

That's not entirely the case. For example, `LlamaForSequenceClassification` and `LlamaForQuestionAnswering` have `_tp_plan=None` (at its top level), while `LlamaForCausalLM` has a `_tp_plan = {"lm_head": ...}`.

Reply:
Yep, indeed.  I thiught we should enforce TP plan definition for all classes to avoid user errors but its fine like this!